code/game/objects/items/stacks/stack.dm

```diff
@@ -19,18 +19,20 @@
 	var/cost = 1 // How much energy from storage it costs
 	var/merge_type = null // This path and its children should merge with this stack, defaults to src.type
 
-/obj/item/stack/New(var/loc, var/amount=null)
+/obj/item/stack/Initialize(mapload, amount=null)
 	..()
-	if (amount)
-		src.amount = amount
+	if(amount)
+		amount = amount
```

Comment:
This src.amount was actually necessary since there's an argument named amount and a var named amount on the stack itself. Either rename the argument or readd the src.amount.